indirect_value.h

```diff
@@ -110,8 +130,22 @@ class ISOCPP_P1950_EMPTY_BASES indirect_value
   explicit indirect_value(std::in_place_t, Ts&&... ts)
       : ptr_(new T(std::forward<Ts>(ts)...)) {}
 
+  template <class U, class = std::enable_if_t<std::is_same_v<T, U> &&
+      std::is_default_constructible_v<C> &&
+      not std::is_pointer_v<C> &&
+      std::is_default_constructible_v<D> &&
+      not std::is_pointer_v<D>>>
+  explicit indirect_value(U* u) noexcept
+      : copy_base(C{}), delete_base(D{}), ptr_(u) {}
+
+  template <class U, class = std::enable_if_t<std::is_same_v<T, U> &&
+      std::is_default_constructible_v<D> &&
+      not std::is_pointer_v<D>>>
+  indirect_value(U* u, C c) noexcept
+      : copy_base(std::move(c)), delete_base(D{}), ptr_(u) {}
+
   template <class U, class = std::enable_if_t<std::is_same_v<T, U>>>
-  explicit indirect_value(U* u, C c = C{}, D d = D{}) noexcept
+  indirect_value(U* u, C c, D d) noexcept
```

Comment:
I think we want to keep `explicit` here.

Without `explicit` a pointer and two functors passed as `{p, a, b}` might get converted into an `indirect_value`. That could be very bad albeit unlikely as it has ownership implications.

`explicit` is not always needed {pointer, size} can be interpreted as a `std::span<const T>` with no ill-effect.